pkg/scheduler/framework/preemption/preemption.go

```diff
@@ -401,6 +461,71 @@ func (ev *Evaluator) prepareCandidate(ctx context.Context, c Candidate, pod *v1.
 	return nil
 }
 
+// prepareCandidateAsync triggers a goroutine for some preparation work:
+// - Evict the victim pods
+// - Reject the victim pods if they are in waitingPod map
+// - Clear the low-priority pods' nominatedNodeName status if needed
+// The Pod won't be retried until the goroutine triggered here completes.
+//
+// See http://kep.k8s.io/4832 for how the async preemption works.
+func (ev *Evaluator) prepareCandidateAsync(c Candidate, pod *v1.Pod, pluginName string) {
+	metrics.PreemptionVictims.Observe(float64(len(c.Victims().Pods)))
+
+	// Intentionally create a new context, not using a ctx from the scheduling cycle, to create ctx,
+	// because this process could continue even after this scheduling cycle finishes.
+	ctx, cancel := context.WithCancel(context.Background())
+	errCh := parallelize.NewErrorChannel()
+	preemptPod := func(index int) {
+		victim := c.Victims().Pods[index]
+		if err := ev.PreemptPod(ctx, c, pod, victim, pluginName); err != nil {
+			errCh.SendErrorWithCancel(err, cancel)
+		}
+	}
+
+	ev.mu.Lock()
+	ev.preempting[pod.UID] = struct{}{}
```

Comment:
```suggestion
	ev.preempting.Insert(pod.UID)
```

nits: It seems that the `preempting` is defined as a set typed.
Is there any policy to use the map operation instead of the set operation?

https://github.com/kubernetes/kubernetes/blob/4c2ebc1e2a49e9477669bd22831fd458c32fae7f/pkg/scheduler/framework/preemption/preemption.go#L136-L138

Reply:
Please use `Insert`

Reply:
Thanks, 

> Is there any kube-scheduler policy using the map operation instead of the set operation?

No no, it was a map type before, someone suggested me using `set` at some point, and I just forgot changing the operation to use methods after the change.